cmake/EthDependencies.cmake

```diff
@@ -34,10 +34,9 @@ set(BOOST_COMPONENTS "filesystem;unit_test_framework;program_options;system")
 if (WIN32)
 	# Boost 1.77 fixes a bug that causes crashes on Windows for some relative paths in --allow-paths.
 	# See https://github.com/boostorg/filesystem/issues/201
-	find_package(Boost 1.77.0 QUIET REQUIRED COMPONENTS ${BOOST_COMPONENTS})
+	find_package(Boost 1.77.0 EXACT QUIET REQUIRED COMPONENTS ${BOOST_COMPONENTS})
```

Comment:
Is there really call for this? How bad are the issues with > 1.78? But well, ok, maybe we can keep it for now, since we have an issue staged for fixing it and can then relax it again...

Reply:
1. Unless you upgrade to visual studio 2022 (2020 is documented as working), you can't even build 1.78
2. If you do upgrade and get everything running - soltest fails due to different behaviour of `boost::filesystem`

Reply:
Yeah, but how bad are the ``boost::filesystem`` failures? Is that mainly technicalities or proper issues? Depending on that, I'd just allow >1.77 here anyways... overrestricting the build system requirements is almost as bad as undersrestricting them...

Reply:
We already have an issue open to bump everything on Windows, I believe Chris P tried it some time ago, then I tried it a couple of months ago, and in general I feel as restricting it to an exact version is the correct move until we fully fix this - but I can certainly remove the `EXACT` from here since no one really builds Solidity from source on Windows anyway :D

Reply:
> IIRC, MSVC2022 kind of works if boost is updated (I think I did it myself, but I don't recall precisely). 
It's the other way around - you need MSVC2022 in order to build boost 1.78 (`b2` )
I'll remove the `EXACT` as you make a good enough point :)

Reply:
I don't think many users would be affected in practice (or at least they may not notice) but it changes compiler behavior depending on which Boost version you compile with so I'd still consider it highly undesirable. In corner cases you could get different metadata between platforms for the same code, depending on how you build the compiler.

From what we established in #13434, the normalization on newer versions is different for paths starting with `//`. And in some cases you might get stuff normalized to `\path` instead of `\path\.` as before. The latter probably does not matter much for us (I'd have to make sure though) but you could run into the former if you're sloppy about your paths.

Overall the fact that a part of the language depends on aspects of `filesystem` that can and do change between versions (and implementations - `std::filesystem` has slightly different rules as well) is messy. You think you're saving yourself a lot of hassle by not having to write all that FS code but then you have to have a very detailed test suite to make sure behavior stays consistent. And in our case not all of our filesystem use is covered with tests so I would not be surprised if there are more things like this that we just do not see.

Reply:
I think that `EXACT` should stay. Build with 1.78 simply gives you the compiler that does not behave correctly, even if only in corner cases. I mean, as a user, why would you want to use version known to be even slightly broken if you can get a perfectly good one with 1.77?

The fact that tests won't pass will give you hint that something's wrong but that's too late. User is likely to ignore it after going through all the hassle of installing all dependencies and getting it to build. It's much more helpful to fail immediately.

Reply:
I'll put the `EXACT` back since I have to rebase anyway.

Reply:
Ultimately, bumping to 1.78 requires MSVC2022, and if both are bumped, our builds do not succeed (bytecode compare if remember correctly), which I believe is not good enough for us to recommend it in the docs.

Reply:
Yes, but also the other way round. I'm categorically against fixing versions in cmake.

Reply:
Well, maybe I'll just try to create a proper fix for the boost changes tomorrow. They are not actually that big so maybe we can avoid this problem by just making it work with both 1.77 and 1.78.

Reply:
> Well, maybe I'll just try to create a proper fix for the boost changes tomorrow. They are not actually that big so maybe we can avoid this problem by just making it work with both 1.77 and 1.78.

This was supposed to be a 5 minute doc change PR, and it's already turned into 36 comment discussion. I honestly don't think this should be our priority at the moment, especially seeing as we haven't released in a while.

Reply:
We shouldn't even have started messing with the unrelated windows case here in the first place. But the other thread https://github.com/ethereum/solidity/pull/13874#discussion_r1071488393 here is rather important .